### PR TITLE
Search field is focused when you open the Models Library dialog

### PR DESCRIPTION
**Pull request by Idloj**
for the first time. Subsequent openings open in the last state (probably a model selected). If the search field should be focused on every opening of the Models Library, just let me know.

### CODE REVIEW

File: src/main/org/nlogo/app/ModelsLibraryDialog.java [hunk 1]
```diff
@@ -32,11 +34,6 @@ public static String open(java.awt.Frame parent)
       me = new ModelsLibraryDialog(parent);
       me.tree.setSelectionRow(0);
     }
-    // not sure I really understand why this should be necessary
-    // (it isn't on Windows), but at least on Macs, this ensures
-    // that the tree gets the keyboard focus when the dialog opens
-    // - ST 12/17/04
-    me.setFocusable(false);
```
Comment (mrerrormessage) [inline src/main/org/nlogo/app/ModelsLibraryDialog.java]: @SethTisue , I see that this comment is well over ten years old, but do you remember whether it was particularly important to anyone that the file tree get keyboard focus on dialog open? It seems like the search box is a much more natural place to put the focus, but perhaps that didn't exist in early iterations of this class?

**Comment by mrerrormessage** [comment]
This PR is really a design change. I happen to agree with it, but I want to make sure that I understand the rationale for the prior design (as much as possible) before actually merging this.

File: src/main/org/nlogo/app/ModelsLibraryDialog.java (hunk 1 above)
Comment (SethTisue) [inline src/main/org/nlogo/app/ModelsLibraryDialog.java]: I'm afraid I don't remember anything about this

I do like using the arrow keys to navigate the tree, but I'm fine w/ the change.

**Comment by nicolaspayette** [comment]
As a guy that spends a lot of time opening models, I would welcome such a change.

**Comment by qiemem** [comment]
How about the following:

Search box focused by default (as is done here), but then up/down arrows focus model tree (and perform their action there).

Additional features that should probably be added (these are just basic UI things that are common):

- Tab switches focus between tree and search box
- cmd/ctrl-f focuses search box, with the text selected

**Comment by SethTisue** [comment]
> Tab switches focus between tree and search box

That would be great, and would remove any slight hesitancy I might have about the default-focus-location change.

**Comment by Idloj** [comment]
@qiemem, I hope to do these soon, but I fear I don't understand "cmd/ctrl-f focuses search box, with the text selected".

**Comment by qiemem** [comment]
> I fear I don't understand "cmd/ctrl-f focuses search box, with the text selected".

In most application, hitting ctrl-f (in Windows/Linux) brings up a search bar with the text of the search bar selected. Selecting the full text of the search bar is nice because it lets you quickly replace the previous thing you searched for with something new. In OS X, the hotkey is cmd-f instead of ctrl-f. There should be a way to handle the cmd vs ctrl thing automatically, as other NetLogo hotkeys change depending on OS.

I'm suggesting that we replicate this feature in the models library selection dialog. So hitting ctrl-f (on Windows/Linux) or cmd-f (on OS X) focus on the search box (so that when you type, it types in the search box) and selects all the text in the search box (to make it easy to replace your previous search).

**Comment by Idloj** [comment]
> Tab switches focus between tree and search box
Currently, Tab cycles through all the components, so if you want to switch back-and-forth you would press Tab and Shift+Tab. If this is changed so Tab cycles just through the tree and the search box, how would you move to another component (e.g., "Go to User Community Models web page") using keyboard?

**Comment by qiemem** [comment]
Tab and shift tab don't do anything for me at the moment.

**Comment by mrerrormessage** [comment]
@Idloj , it looks like this PR has fallen behind the current hexy branch and can't be merged in without some work. Are you interested in continuing to work on it? If not, I'll go ahead and close it.

**Comment by Idloj** [comment]
I can make it adhere to the current directory structure if that's what you mean

**Comment by Idloj** [comment]
I'm going to squash this into one commit, OK?

**Comment by mrerrormessage** [comment]
sounds good

**Comment by Idloj** [comment]
@mrerrormessage Is there anything I need to do that stops the PR from being merged?

**Comment by mrerrormessage** [comment]
This looks good and ready to merge as soon as it merges cleanly with `hexy`. You can merge hexy into it or rebase it on top of hexy, but other than that it looks good to merge.

**Comment by Idloj** [comment]
Rebased

**Pull request merged**